### DISCUSSION TITLE
Why throw in `onSuccess` doesn't trigger `onError` in `useQuery` but DOES in `useMutation`?

### DISCUSSION THREAD

**Discussion by nvmnghia** [TanStack/query#3751] [General]
Example:

https://codesandbox.io/s/broken-lake-idleic?file=/src/App.js

I have a vague feeling that this is because `mutate()` in `useMutation()` is triggered inside a callback, which *swallow* the error.

**Comment by clicktodev** [comment]
I thought this might be related to #3862 but the problem still exists after upgrading to v4. Perhaps @TkDodo can chime in.

**Comment by TkDodo** [comment]
yeah I think this is not implemented in queries. code would be here, I'd accept a PR that fixes this:

https://github.com/TanStack/query/blob/46ad29a57163a768fdc05f1a12ae2cc5c69b2470/packages/query-core/src/queryObserver.ts#L677-L683
